engine.py

```diff
@@ -501,24 +501,113 @@ def _safe_path_join(self, *args):
+        :param base_version:
+        :return:
+        """
+
+        def version_less_or_equal(v1, v2):
+            # A recursive function that everytime it need to check a lower level in the version tuples
+            # it calls itself again with a sliced the version tuple excluding the first number in the tuple.
+            if v1[0] != "x" and v2[0] != "x":
+                if int(v1[0]) > int(v2[0]):
+                    # We have two numbers and the check number is greater than the base so
+                    # no match, and no need to check lower levels.
+                    return False
+                if int(v1[0]) < int(v2[0]):
+                    # We have two numbers and the check number is less than the base so
+                    # this is a match no need to check deeper
+                    return True
+
+            # Either one of the numbers is "x" or the numbers are the same
+            if len(v1) > 1:
+                # We have more levels so, check another level deeper.
+                # slice off this level so that it will check the next level.
+                return version_less_or_equal(v1[1:], v2[1:])
+            else:
+                # No more levels to check this must be a match since x means any version number.
+                # or the numbers are the same (though this is unlikely as this would mean two identical folders.)
+                return True
+
+        return version_less_or_equal(check_version, base_version)
+
+    def _load_app_otls(self, oplibrary_path):
         """
         Load any OTLs provided by applications.
 
         Look in any application folder for a otls subdirectory and load any .otl
         file from there.
+        :param oplibrary_path: A temporary path that can be used to install otls to
         """
+
+        def install_otls_from_dir(root_path):
+            for filename in os.listdir(root_path):
+                full_path = self._safe_path_join(root_path, filename)
+                if os.path.splitext(filename)[-1] == ".otl":
+                    path = full_path.replace(os.path.sep, "/")
+                    hou.hda.installFile(path, oplibrary_path, True)
+
         for app in self.apps.values():
             otl_path = self._safe_path_join(app.disk_location, "otls")
-            if not os.path.exists(otl_path):
+            for a_path in self._get_otl_paths(otl_path):
+                install_otls_from_dir(a_path)
+
+    def _get_otl_paths(self, otl_path):
+        """
+        For a given app otl folder, return a list of otl folders.
+        The list will always contain the root otl_path, but also may contain a sub
+        version folder if found.
+        It will also check to see if there are any Houdini version folder inside
+        following the format of "v18.0.0". "x" characters can be used instead of
+        a number to represent any number.
+        """
+        if not os.path.exists(otl_path):
+            return []
+
+        otl_paths = [otl_path]
```

Comment:
Adds the otl folder to the list of folders to import otls from, which maintains backwards compatibility.